xmtp_mls/src/builder.rs

```diff
@@ -107,21 +101,36 @@ where
             .ok_or(ClientBuilderError::MissingParameter {
                 parameter: "api_client",
             })?;
+        let network = self.network;
         let store = self.store.take().unwrap_or_default();
-        // Fetch the Identity based upon the identity strategy.
-        let identity = match self.identity_strategy {
-            IdentityStrategy::CachedOnly(_) => {
-                // TODO: persistence/retrieval
-                unimplemented!()
-            }
-            IdentityStrategy::CreateIfNotFound(owner) => {
-                // TODO: persistence/retrieval
-                Identity::new(CIPHERSUITE, &XmtpOpenMlsProvider::default(), &owner)?
+        let provider = XmtpOpenMlsProvider::new(&store);
+        let identity = self.initialize_identity(&store, &provider)?;
+        Ok(Client::new(api_client, network, identity, store))
+    }
+
+    fn initialize_identity(
+        self,
+        store: &EncryptedMessageStore,
+        provider: &XmtpOpenMlsProvider,
+    ) -> Result<Identity, ClientBuilderError> {
+        let conn = &mut store.conn()?;
+        let identity_option: Option<Identity> = conn.fetch(())?.map(|i: StoredIdentity| i.into());
+        match self.identity_strategy {
```

Comment:
could possibly move this into an `impl` on `IdentityStrategy`, if it's used more often